front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```
@@ -449,22 +451,59 @@ const handleCreateTimetableItem = async (
         }
       : undefined,
   };
-  const newTimetableItems = await dispatch(
-    osrdEditoastApi.endpoints.postTimetableByIdPacedTrains.initiate({
-      id: timetableId,
-      body: [pacedTrain],
-    })
-  ).unwrap();
-  if (newTimetableItems.length === 0) {
-    throw new Error('Failed to create paced train');
+  if (trainrun.direction === 'one_way') {
+    // This case should no longer happen in practice, as by default nge should create trains as round trips
+    const newTimetableItems = await dispatch(
+      osrdEditoastApi.endpoints.postTimetableByIdPacedTrains.initiate({
+        id: timetableId,
+        body: [pacedTrain],
+      })
+    ).unwrap();
+
+    if (newTimetableItems.length !== 1) {
+      throw new Error('Failed to create paced train');
+    }
+    const newPacedTrain: PacedTrainWithPacedTrainId = {
+      ...newTimetableItems[0],
+      id: formatEditoastIdToPacedTrainId(newTimetableItems[0].id),
+    };
+    state.timetableItemIdByNgeId.set(trainrun.id, [newPacedTrain.id, null]);
+    addUpsertedTimetableItems([newPacedTrain]);
+  } else {
+    const { path: returnPath, schedule: returnSchedule } = generatePathAndSchedule(
+      trainrunSections,
+      netzgrafikDto.nodes,
+      startDate,
+      TRAINRUN_DIRECTIONS.BACKWARD
+    );
+    const returnPacedTrain = { ...pacedTrain, path: returnPath, schedule: returnSchedule };
+
+    const newTimetableItems = await dispatch(
+      osrdEditoastApi.endpoints.postTimetableByIdPacedTrains.initiate({
+        id: timetableId,
+        body: [pacedTrain, returnPacedTrain],
+      })
+    ).unwrap();
+    if (newTimetableItems.length !== 2) {
+      throw new Error('Failed to create paced train with return trip');
+    }
+    await dispatch(
+      osrdEditoastApi.endpoints.postRoundTripsPacedTrains.initiate({
+        roundTrips: { round_trips: [[newTimetableItems[0].id, newTimetableItems[1].id]] },
+      })
+    );
```

We should `unwrap()` here to ensure errors aren't silently ignored (ie, throw an error instead of doing nothing).

---

Sure!